test/test_parameter.h

```diff
@@ -0,0 +1,59 @@
+/*
+    Copyright (c) 2013, Taiga Nomi and the respective contributors
+    All rights reserved.
+
+    Use of this source code is governed by a BSD-style license that can be found
+    in the LICENSE file.
+*/
+#pragma once
+#include "gtest/gtest.h"
+#include "testhelper.h"
+#include "tiny_dnn/tiny_dnn.h"
+
+namespace tiny_dnn {
+
+TEST(parameter, init) {
+  parameter p(3, 3, 1, 1, parameter_type::weight, true);
+
+  ASSERT_EQ(p.shape().width_, 3u);
+  ASSERT_EQ(p.shape().height_, 3u);
+  ASSERT_EQ(p.shape().depth_, 1u);
+  ASSERT_EQ(p.size(), 9u);
+  ASSERT_EQ(p.type(), parameter_type::weight);
+
+  ASSERT_TRUE(p.is_trainable());
+}
+
+TEST(parameter, getter_setter) {
+  parameter p(4, 1, 1, 1, parameter_type::bias, false);
+
+  xt::xarray<float_t> xr = {1.0, 2.0, 3.0, 4.0};
```

Comment:
I'd would go for a simple vector. We should avoid to use `xt::` outside of `Tensor`